### DISCUSSION TITLE
First Footfall Event / Variable?

### DISCUSSION THREAD

**Discussion by geogenic** [EDCD/EDDI#2329] [Needs thought]
Apologies in advance if this has already been brought up, but I was wondering if there's a way to implement a first footfall event or variable in EDDI, or if there might be a workaround.

For the time being, I've created a script that reads out a monologue during the `disembark` event if the body hasn't been discovered already, but there's only "first discovery" variables for `alreadymapped` and `alreadydiscovered`.

Side note, huge thanks to all the devs that have put work into EDDI, it's become indispensable to me!

**Comment by Tkael** [comment]
I'm not sure if there's an event for that. Maybe a `CodexEntry` event? If you have a player journal recorded during a first footfall then I can inspect it and see if there's a signal that we could turn into an event. o7
Player journals are saved to `[Saved Games]/Frontier Developments/Elite Dangerous`.

**Reply by geogenic** [reply]
Here's a journal entry that has a couple first footfalls. It looks like there's no event for it, or anything special that happens during the disembark event. It seems strange that FDEV would include the `alreadymapped` and `alreadydiscovered` parameters, but not something similar for first footfall since all three are listed in the system map. As for a codex entry event, is that already implemented in EDDI? If not, that would be super awesome for announcing the names of species or geological sites scanned while on foot.

Journal: [Journal.220302195800.01.log](https://github.com/EDCD/EDDI/files/8178743/Journal.220302195800.01.log)

**Reply by geogenic** [reply]
There is in fact a `CodexEntry` event in the journals, as well as a `ScanOrganic` event that lists the species name scanned. If there was a way to reference the values in `ScanOrganic` in EDDI, that would be awesome.

**Reply by zajc3w** [reply]
ED CoPilot acesses first footfall somehow, i'm getting voice prompt when getting first footfall. It;s bugged anyway, yesterday i got first footfall bonus on one of billionaire avenue systems.

**Reply by Tkael** [reply]
There is no player journal event for first footfall. ED CoPilot is closed source so I can't inspect their code to see what they are detecting.
